connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -138,6 +139,7 @@ public WorkerSinkTask(ConnectorTaskId id,
         this.sinkTaskMetricsGroup.recordOffsetSequenceNumber(commitSeqno);
         this.consumer = consumer;
         this.isTopicTrackingEnabled = workerConfig.getBoolean(TOPIC_TRACKING_ENABLE_CONFIG);
+        this.taskStopped = false;
```

Comment:
Could it be replaced by ```isStopping()```? It seems to me both flags are similar and we don't need to add more duplicate.

Reply:
I originally considered this approach, but it won't work for cases where the task stops on its own due to failure instead of being stopped externally by the worker.

Reply:
> it won't work for cases where the task stops on its own due to failure instead of being stopped externally by the worker.

thanks for the explanation. The case you mentioned is that we don't call ```onFailure``` before closing task so ```isStopping``` still return true. Could you add unit test for that case?

Reply:
Given the complexity of the unit tests for the `WorkerSinkTask` class and the guarantees we get from the coverage of the existing tests, I'm not sure it's really worth the effort. The code path that's modified in this PR is agnostic about the cause of shutdown for the task and we won't really get any more coverage by simulating a shutdown triggered by an exception from the `SinkTask` instance instead of external request from the `Worker` instance.